### ISSUE TITLE
Instance reference object property

### ISSUE THREAD

**Issue by Gamer-XP** [YoYoGames/GameMaker-Feature-Requests#1571]
### Is your feature request related to a problem?

It's pretty common situation for me where I want to reference another instance in the room. Doing it by their ids is fine, but it's not very convenient and lacks visual feedback.

### Describe the solution you'd like

Add "Instance reference" field type. It should allow to select another instance in the room as a target. You click [Select Target] button, and it should highlight options on the scene you can select. Clicking on them will write them down to the field.

Possible extra settings are:
1. Object type filter
2. Tag filter

Also may be good to draw some sort of line between 2 objects, indicating they are connected by property.

**Comment by nullbunny** [comment]
I love this. Array of Instance Refs would also be useful.

**Comment by Gamer-XP** [comment]
Yes. I've made a separate request for an array. They are supposed to work together
https://github.com/YoYoGames/GameMaker-Feature-Requests/issues/1570

**Comment by Hyomoto** [comment]
A picture is worth a thousand words, and in this case perhaps a thousand lines of code,
![image](https://github.com/YoYoGames/GameMaker-Feature-Requests/assets/6281477/d1addc31-5c37-4569-9824-a66df568d1c6)

The issue that all of these requests are seeking to solve is that the room editor lacks a great way to tie together instances into a logical chain.  This can be done with code, but it's suboptimal to view that data this way.  While how complex this must be is up for debate, the bare minimum pieces that would make it useful are: being able to select one, or more, instances as a variable definition type, and being able to visualize those connections.  This is a fairly major omission from the room editor and has been for years as we wait on a "new room editor" that's just two decades around the corner.

Thus, as a stopgap, this would be a fairly low-hanging fruit in terms of overall utility.
